src/Sarif.WorkItems/SarifWorkItemContext.cs

```diff
@@ -14,11 +15,20 @@ public class SarifWorkItemContext : PropertiesDictionary
 
         internal void InitializeFromLog(SarifLog sarifLog)
         {
-            // TODOD: Initialize source and non-source-controlled files here,
-            //        as determined by scraping all URLs associated with results.
-            // 
-            //        https://github.com/microsoft/sarif-sdk/issues/1752
+            var visitor = new ExtractAllArtifactLocationsVisitor();
+            visitor.VisitSarifLog(sarifLog);
+            foreach (ArtifactLocation location in visitor.AllArtifactLocations)
+            {
+                if(location?.Uri != null)
+                {
+                    LocationUri = location.Uri;
+                    break;
+                }
+            }
         }
+
+        public Uri LocationUri { get; private set; }
```

Comment:
>LocationUri [](start = 19, length = 11)

Why is this in the context and not the SARIF work item model?

Reply:
To me, the work item model contains information the work item filer needs to know about, and the context contains information that the plugins might use to determine that information.  I do not anticipate the filer needing this location directly, it instead uses the area path and owner information that the plugin pulls using this URI location.

I'm open to changing my mind on this, but that's where I've been drawing the line of distinction.

---
In reply to: [390639507](https://github.com/microsoft/sarif-sdk/pull/1808#discussion_r390639507) [](ancestors = 390639507)

Reply:
Discussed offline, moving to SarifWorkItemModel for now, may eventually rethink need for that class and determine clear separation of parts as we determine exactly what supporting\configuration data is necessary.

---
In reply to: [390643075](https://github.com/microsoft/sarif-sdk/pull/1808#discussion_r390643075) [](ancestors = 390643075,390639507)